doc: adds assert doc for strict mode with pointer to strict equality (#30486)
nodejs/node: doc/api/assert.md

> In `strict` mode, `assert` functions use the comparison in the corresponding

Reviewer: Maybe just write:

```suggestion
 In `strict` mode (not to be confused with `"use strict"`), `assert` functions ...
```

strict functions. For example, [`assert.deepEqual()`][] will behave like

> [`assert.deepStrictEqual()`][]. This uses the [Strict Equality Comparison][]

Reviewer: That is actually not correct. The `strict` part just refers to a more precise comparison, not to the used algorithm.
The actual comparison steps are listed at the corresponding entry.

Author: Got it! I'm removing this change as it seems confusing. I'll be keeping the other suggestion down below.
Thanks. 👍
